src/audio/asrc/asrc.c

```diff
@@ -517,15 +359,13 @@ static int asrc_params(struct processing_module *mod)
 	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
 				source_list);
 
-#if CONFIG_IPC_MAJOR_4
 	/* update the source/sink buffer formats. Sink rate will be modified below */
-	ipc4_update_buffer_format(sourceb, &cd->ipc_config.base.audio_fmt);
-	ipc4_update_buffer_format(sinkb, &cd->ipc_config.base.audio_fmt);
-#endif
+	update_buffer_format(sourceb, cd);
+	update_buffer_format(sinkb, cd);
 
 	/* Don't change sink rate if value from IPC is 0 (auto detect) */
-	if (asrc_get_sink_rate(&cd->ipc_config))
-		audio_stream_set_rate(&sinkb->stream, asrc_get_sink_rate(&cd->ipc_config));
+	if (cd->sink_rate)
+		audio_stream_set_rate(&sinkb->stream, cd->sink_rate);
```

Comment:
no need change, also PR title is not code clean, it is code reorganize.

Reply:
yeah - nothing wrong with the cleanup, we should just put into into the next patch so we can bisect if an issue is introduced.